### PR TITLE
Fixes for the rest of the CLI tests on Treasure Island.

### PR DESCRIPTION
**Pull request by jMyles**

### CODE REVIEW

File: nucypher/network/nodes.py [hunk 1]
```diff
@@ -759,30 +767,31 @@ def learn_from_teacher_node(self, eager=False):
...
-                              f'{sprout} is not bonded to a Staker.')
-
-            except sprout.Invalidsprout:
-                self.log.warn(sprout.invalid_metadata_message.format(sprout))
-
-            except sprout.SuspiciousActivity:
-                message = f"Suspicious Activity: Discovered sprout with bad signature: {sprout}." \
-                          f"Propagated by: {current_teacher}"
-                self.log.warn(message)
+            # TODO: This whole section is weird; sprouts down have any of these things.
```
Comment (KPrasch) [inline nucypher/network/nodes.py:770]: Woah - I guess this logic needs complete relocation?

File: tests/acceptance/cli/lifecycle.py [hunk 2]
```diff
@@ -90,46 +71,13 @@ def fetch_bob_public_keys(self) -> BobPublicKeys:
...
-
-
-def _cli_lifecycle(click_runner,
-                   testerchain,
-                   random_policy_label,
-                   ursulas,
-                   custom_filepath,
-                   custom_filepath_2,
-                   registry_filepath=None):
+def run_entire_cli_lifecycle(click_runner,
```
Comment (KPrasch) [inline tests/acceptance/cli/lifecycle.py:74]: :+1:

File: tests/acceptance/cli/lifecycle.py [hunk 3]
```diff
@@ -396,9 +347,9 @@ def _bob_retrieves(_grant_result):
 
     # Run the Callbacks
     d = threads.deferToThread(enrico_encrypts)  # scene 4
-    d.addCallback(_alice_decrypts)              # scene 5 (uncertainty)
-    d.addCallback(_run_teacher)                 # scene 6 (preamble)
-    d.addCallback(_grant)                       # scene 7
-    d.addCallback(_bob_retrieves)               # scene 8
```
Comment (KPrasch) [inline tests/acceptance/cli/lifecycle.py:402]: precommit reformatted this?

File: tests/acceptance/cli/ursula/test_run_ursula.py [hunk 4]
```diff
@@ -147,7 +147,7 @@ def run_ursula(teacher_uri):
 
     yield d
 
-
+@pytest.mark.skip("Let's put this on ice until we get 2099 and Treasure Island working together.")
```
Comment (KPrasch) [inline tests/acceptance/cli/ursula/test_run_ursula.py:150]: How do we track this one?

File: nucypher/characters/base.py [hunk 5]
```diff
@@ -161,10 +153,6 @@ def __init__(self,
         else:
             self._crypto_power = CryptoPower(power_ups=self._default_crypto_powerups)
 
-        # Fleet and Blockchain Connection (Everyone)
```
Comment (KPrasch) [inline nucypher/characters/base.py:164]: :+1: - in the spirit of keeping defaults out of characters.  What in a sense "allowed" this change to occur now?

File: tests/acceptance/cli/test_federated_cli_lifecycle.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+import pytest_twisted as pt
+
+from tests.acceptance.cli.lifecycle import run_entire_cli_lifecycle
+
+
+@pt.inlineCallbacks
+def test_federated_cli_lifecycle(click_runner,
```
Comment (derekpierre) [inline tests/acceptance/cli/test_federated_cli_lifecycle.py:23]: Any reason why we need the `testerchain` fixture for the federated tests? Can we pass `testerchain = None` to the `run_entire_cli_lifecycle` method, or is there something precluding us from doing so?

File: tests/integration/learning/test_discovery_phases.py [hunk 7]
```diff
@@ -193,3 +202,7 @@ def find_successful_responses(map_publication_responses):
         # But with debuggers and other processes running on laptops, we give a little leeway.
         assert initial_blocking_duration.total_seconds() < 3
         assert complete_distribution_time.total_seconds() < 10
+
+        # Takes about .8 on jMyles' laptop; still the bulk of init time.
+        # Assuming this is 10 percent of the fleet, that's another 8 second savings for first policy enactment.
+        assert maturation_time.total_seconds() < 2
```
Comment (derekpierre) [inline tests/integration/learning/test_discovery_phases.py:208]: `pre-commit` hooks not running?

File: nucypher/characters/base.py [hunk 8]
```diff
@@ -555,3 +543,6 @@ def make_cli_controller(self, crash_on_error: bool = False):
 
         self.controller = controller
         return controller
+
+    def disenchant(self):
```
Comment (derekpierre) [inline nucypher/characters/base.py:547]: Might be picking nits, but I'm not sure I'm sold on this name - seems overly complicated 😕 . IN this spirit of your wordplay - `disassociate` or `disconnect`?

Comment (jMyles) [reply nucypher/characters/base.py:547]: Haha, yeah, it's not a very good word.

I think that the real thing is to just make a single `stop` and tell people they can't restart.

Comment (KPrasch) [reply nucypher/characters/base.py:547]: I'm really enjoying the character animation metaphor unfolding here.  Perhaps simply `stop` will do but I don't mind `disenchant`... (what happens at the end of animation?) its not as flowery as "bow out" for example :-)

File: nucypher/network/nodes.py (hunk 1 above)
Comment (jMyles) [reply nucypher/network/nodes.py:770]: I'm not sure.  It needs either:

* Relocation
* New imports
* Abandonment altogether

How important is it?  I guess probably somewhat.

File: nucypher/cli/utils.py [hunk 9]
```diff
@@ -74,7 +74,7 @@ def make_cli_character(character_config,
 
     # Handle Teachers
     # TODO: Is this still relevant?  Is it better to DRY this up by doing it later?
-    teacher_nodes = list()
+    sage_nodes = list()
```
Comment (cygnusv) [inline nucypher/cli/utils.py:77]: 🧙

File: nucypher/cli/utils.py [hunk 10]
```diff
@@ -89,7 +89,15 @@ def make_cli_character(character_config,
     #
 
     # Produce Character
-    CHARACTER = character_config(known_nodes=teacher_nodes,
+    if teacher_uri:
+        maybe_sage_node = character_config.known_node_class.from_teacher_uri(teacher_uri=teacher_uri,
+                         min_stake=0,  # TODO: Where to get this?
+                         federated_only=character_config.federated_only,
+                         network_middleware=character_config.network_middleware,
+                         registry=character_config.registry)
```
Comment (cygnusv) [inline nucypher/cli/utils.py:97]: The formatting of these lines seems a bit funky

File: nucypher/config/node.py [hunk 11]
```diff
@@ -61,6 +63,9 @@ class CharacterConfiguration(BaseConfiguration):
     DEFAULT_NETWORK_MIDDLEWARE = RestMiddleware
     TEMP_CONFIGURATION_DIR_PREFIX = 'tmp-nucypher'
 
+    # When we begin to support other threshold schemes, this will be one of the concepts that makes us want a factory.
```
Comment (cygnusv) [inline nucypher/config/node.py:66]: Seems related to #571? If so:
```suggestion
    # When we begin to support other threshold schemes, this will be one of the concepts that makes us want a factory - #571.
```

File: nucypher/network/nodes.py [hunk 12]
```diff
@@ -243,25 +243,26 @@ def load_seednodes(self, read_storage: bool = True):
...
-                                                       federated_only=self.federated_only,
-                                                       network_middleware=self.network_middleware,
-                                                       registry=self.registry)
-            except NodeSeemsToBeDown:
-                self.unresponsive_seed_nodes.add(uri)
-            else:
-                if maybe_sage_node is UNKNOWN_VERSION:
-                    continue
+        if self.learning_domains:
+            canonical_sage_uris = self.network_middleware.TEACHER_NODES.get(tuple(self.learning_domains)[0], ())  # TODO: Are we done with multiple domains?
```
Comment (cygnusv) [inline nucypher/network/nodes.py:249]:
```suggestion
            canonical_sage_uris = self.network_middleware.TEACHER_NODES.get(tuple(self.learning_domains)[0], ())  # TODO: Are we done with multiple domains? - #1580
```

File: tests/acceptance/cli/lifecycle.py [hunk 13]
```diff
@@ -1,31 +1,13 @@
...
- nucypher is distributed in the hope that it will be useful,
- but WITHOUT ANY WARRANTY; without even the implied warranty of
- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- GNU Affero General Public License for more details.
-
- You should have received a copy of the GNU Affero General Public License
- along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
-"""
-
+import datetime
```
Comment (cygnusv) [inline tests/acceptance/cli/lifecycle.py:1]: What's the motivation for removing the license notice? Isn't that something the pre-commit hooks should complain about?

File: tests/acceptance/cli/lifecycle.py [hunk 14]
```diff
@@ -90,46 +71,13 @@ def fetch_bob_public_keys(self) -> BobPublicKeys:
...
-
-def _cli_lifecycle(click_runner,
-                   testerchain,
-                   random_policy_label,
-                   ursulas,
-                   custom_filepath,
-                   custom_filepath_2,
-                   registry_filepath=None):
+def run_entire_cli_lifecycle(click_runner,
+                             testerchain,
```
Comment (cygnusv) [inline tests/acceptance/cli/lifecycle.py:75]: In relation with @derekpierre's comment below, it seems testerchain is only used in the decentralized test to get alice and bob accounts, so maybe you can get rid of it altogether.

File: tests/acceptance/cli/test_federated_cli_lifecycle.py (hunk 6 above)
Comment (cygnusv) [reply tests/acceptance/cli/test_federated_cli_lifecycle.py:23]: Good eye @derekpierre !

File: tests/acceptance/cli/ursula/test_run_ursula.py (hunk 4 above)
Comment (jMyles) [reply tests/acceptance/cli/ursula/test_run_ursula.py:150]: I have the mental bandwidth to keep it in mind for the moment.  As we work on a merge involving both this and @tuxxy's DB work, we can open an issue if needed.

File: nucypher/characters/base.py (hunk 5 above)
Comment (jMyles) [reply nucypher/characters/base.py:164]: Because we no longer load seednoes prior to init time; the entire fail fast logic is now init time.

File: tests/acceptance/cli/lifecycle.py (hunk 13 above)
Comment (jMyles) [reply tests/acceptance/cli/lifecycle.py:1]: Mistake.
